api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -201,32 +192,40 @@ public void updateProfileForUser(DbUser user, Profile updatedProfile, Profile pr
     user.setDemographicSurvey(dbDemographicSurvey);
     user.setLastModifiedTime(now);
 
-    updateInstitutionalAffiliations(updatedProfile, user);
     boolean requireInstitutionalVerification =
         workbenchConfigProvider.get().featureFlags.requireInstitutionalVerification;
     if (requireInstitutionalVerification) {
       validateInstitutionalAffiliation(updatedProfile);
+    } else {
+      updateInstitutionalAffiliations(updatedProfile, user);
     }
 
-    userService.updateUserWithConflictHandling(user);
+    user = userService.updateUserWithConflictHandling(user);
+
     if (requireInstitutionalVerification) {
-      DbVerifiedInstitutionalAffiliation updatedDbVerifiedAffiliation =
-          verifiedInstitutionalAffiliationMapper.modelToDbWithoutUser(
-              updatedProfile.getVerifiedInstitutionalAffiliation(), institutionService);
-      updatedDbVerifiedAffiliation.setUser(user);
-      Optional<DbVerifiedInstitutionalAffiliation> dbVerifiedAffiliation =
-          verifiedInstitutionalAffiliationDao.findFirstByUser(user);
-      dbVerifiedAffiliation.ifPresent(
-          verifiedInstitutionalAffiliation ->
-              updatedDbVerifiedAffiliation.setVerifiedInstitutionalAffiliationId(
-                  verifiedInstitutionalAffiliation.getVerifiedInstitutionalAffiliationId()));
-      this.verifiedInstitutionalAffiliationDao.save(updatedDbVerifiedAffiliation);
+      saveVerifiedInstitutionalAffiliation(user, updatedProfile);
     }
 
     final Profile appliedUpdatedProfile = getProfile(user);
     profileAuditor.fireUpdateAction(previousProfile, appliedUpdatedProfile);
   }
 
+  public void saveVerifiedInstitutionalAffiliation(DbUser user, Profile updatedProfile) {
+    DbVerifiedInstitutionalAffiliation updatedDbVerifiedAffiliation =
+        verifiedInstitutionalAffiliationMapper.modelToDbWithoutUser(
+            updatedProfile.getVerifiedInstitutionalAffiliation(), institutionService);
```

Comment:
nit: `profileToDb()` is fine, since the signature implies that there's no user provided, and there's no DbProfile.

But this feels like going the long way around somehow...